src/System.Private.CoreLib/shared/System/SpanHelpers.cs

```diff
@@ -24,7 +24,7 @@ public static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
                 return;
 
 #if CORECLR && (AMD64 || ARM64)
-            if (byteLength > 4096)
+            if (byteLength > 768)
```

Comment:
Nit: could you add a comment as to how this number was deduced?

Reply:
ran the benchmark for different array sizes:
```
|         Method | arraySize |     Mean |
|--------------- |---------- |---------:|
|          Clear |       500 | 13.00 ns |
| ClearUnaligned |       500 | 11.32 ns |
|          Clear |       600 | 16.99 ns |
| ClearUnaligned |       600 | 16.94 ns |
|          Clear |       700 | 17.21 ns |
| ClearUnaligned |       700 | 23.03 ns |
|          Clear |       800 | 17.60 ns |
| ClearUnaligned |       800 | 18.15 ns |
|          Clear |       900 | 18.02 ns |
| ClearUnaligned |       900 | 19.22 ns |
```
but these results are for the old code `if (byteLength > 4096)` so all of them use `Unsafe.InitBlockUnaligned` and the problem appears after count > 600 on my machine (i8700K Windows10 x64)